ethcore/src/block.rs

```diff
@@ -266,8 +267,9 @@ impl<'x> OpenBlock<'x> {
 		r.block.base.header.set_extra_data(extra_data);
 		r.block.base.header.note_dirty();
 
-		let gas_floor_target = ::std::cmp::max(gas_range_target.0, engine.params().min_gas_limit);
-		engine.populate_from_parent(&mut r.block.base.header, parent, gas_floor_target, gas_range_target.1);
+		let gas_floor_target = cmp::max(gas_range_target.0, engine.params().min_gas_limit);
+		let gas_ceil_target = cmp::max(gas_range_target.1, gas_floor_target);
```

Comment:
It should still work fine even if you setup `ceil < floor`, that's the reason why it says `max(gas_range_target.1, gas_floor_target)` instead of `max(gas_range.target.1, min_gas_limit)`.

In both cases `min_gas_limit > floor > ceil` and `min_gas_limit > ceil > floor` the gas should stay constant with value `min_gas_limit`.